server/internal/store/org.go

```diff
@@ -64,5 +64,6 @@ func (s *S) DeleteOrganization(tenantID, orgID string) error {
 	if res.RowsAffected == 0 {
 		return gorm.ErrRecordNotFound
 	}
-	return nil
+
+	return s.db.Unscoped().Where("organization_id = ?", orgID).Delete(&OrganizationUser{}).Error
```

Comment:
Let's create a DB transaction and delete both `organization` and `organization_user` in a single transaction.